pkg/aggregator/time_sampler_test.go

```diff
@@ -30,23 +30,24 @@ func generateSerieContextKey(serie *metrics.Serie) ckey.ContextKey {
 	return l.Generate(serie.Name, serie.Host, tagset.NewHashingTagsAccumulatorWithTags(serie.Tags))
 }
 
-func testTimeSampler() *TimeSampler {
-	return NewTimeSampler(TimeSamplerID(0), 10, 15*time.Second, metrics.NewMetricSamplePool(32), 100,
-		&serializer.MockSerializer{}, tags.NewStore(false, "test"), flushAndSerializeInParallel{enabled: false})
+func testTimeSampler() (*TimeSampler, *timeSamplerWorker) {
+	sampler := NewTimeSampler(TimeSamplerID(0), 10, tags.NewStore(false, "test"))
+	worker := newTimeSamplerWorker(sampler, 15*time.Second, 100, metrics.NewMetricSamplePool(32), &serializer.MockSerializer{}, flushAndSerializeInParallel{enabled: false})
+	go worker.run()
+	return sampler, worker
```

Comment:
It looks like the tests do not interact with the worker apart from `defer worker.stop()`. Would it make sense to remove it altogether?